torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -1591,7 +1603,7 @@ def test_isend_torch_profiler(self):
             self._test_isend(profiler_ctx=torch_profiler_ctx)
 
         # IRECV
-        @sandcastle_skip_if(BACKEND == "nccl", "Nccl does not support irecv")
+        @require_ucc_for_nccl()
```

Comment:
keep it `sandcastle_skip_if` and just make `require_ucc_for_nccl` be a function returning bool

Reply:
See https://github.com/pytorch/pytorch/pull/69587/files#r850709419